modules/navier_stokes/src/linearfvkernels/LinearFVEnergyAdvection.C

```diff
@@ -0,0 +1,107 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "LinearFVEnergyAdvection.h"
+#include "MooseLinearVariableFV.h"
+#include "NSFVUtils.h"
+#include "NS.h"
+
+registerMooseObject("MooseApp", LinearFVEnergyAdvection);
+
+InputParameters
+LinearFVEnergyAdvection::validParams()
```

Comment:
we should consider with @grmnptr whether we want a kernel for energy when it's the same as scalar and we could just call it scalarAdvection